### PR TITLE
updated to support Mongoid 4.x

### PR DESCRIPTION
**Pull request by brendanstennett**
I updated this gem to support Mongoid 4x. Rather than have branching logic to support different versions, you could just set this version and up to support Mongoid 4x and if someone needs lower, then they should use a different version.

I also noticed you mention you weren't interesting in maintaining the repo anymore.  I use this gem in production so I would take over maintenance if that works with you.

Thanks :)

### CODE REVIEW

**Comment by rehevkor5** [comment]
I can't add you as a contributor or give you access on Rubygems (only @logandk can do that) but I can merge your PR and deploy a new version myself if you give me a little time.

First, I just want to make sure you're aware of https://github.com/logandk/mongoid_denormalize/pull/21 that was never merged. Is there anything in there you'd like to include? Otherwise I am happy to merge this.

**Comment by rehevkor5** [comment]
Or should I merge the other first & allow you to resolve any merge conflicts?

File: lib/mongoid_denormalize.rb [hunk 1]
```diff
@@ -113,7 +117,7 @@ def assign_and_save(obj, assignments, prefix)
     unless attributes_hash.empty?
       # The more succinct update_attributes(changes, :without_protection => true) requires Mongoid 3.0.0, 
       # but we only require 2.1.9
```
Comment (rehevkor5) [inline lib/mongoid_denormalize.rb]: Comment is out of date, can be removed.

**Comment by HuffMoody** [comment]
@rehevkor5 I have seen #21 however it seemed to remove the `:as` options which I thought was a great feature.  This keeps this intact.

I will update the PR with the changes you suggest.

**Comment by logandk** [comment]
@HuffMoody Good work! It's been a while since I last used this project in production, which is why I'm not actively maintaining it. However, @rehevkor5 and a few others have been maintaining it, although they are probably in the same situation, considering the little activity there's been recently.

So if you would like to maintain the gem, I think it would be great. I'll also give you permissions on RubyGems if you send me your email address (which is needed to add a gem owner). I assume this is okay with you, @rehevkor5.

**Comment by brendanstennett** [comment]
@logandk Definitely understandable.

My email for ruby gems is /redacted/

**Comment by logandk** [comment]
@HuffMoody you've been added as a gem owner!

**Comment by brendanstennett** [comment]
@logandk Great! Thanks :)

**Pull request closed without merging**